File: src/main/java/com/flexpoker/config/SecurityConfig.java
package com.flexpoker.config;

import javax.inject.Inject;

import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.builders.WebSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;

import com.flexpoker.login.repository.LoginRepository;
import com.flexpoker.util.PasswordUtils;

@Configuration
@EnableWebSecurity
public class SecurityConfig extends WebSecurityConfigurerAdapter {

    @Inject
    private LoginRepository loginRepository;

    @Override
    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
        auth.userDetailsService(loginRepository).passwordEncoder(PasswordUtils.encoder);
    }

    @Override
    protected void configure(HttpSecurity http) throws Exception {
        http.authorizeRequests().antMatchers("/admin/**")
                .hasRole("ADMIN").antMatchers("/sign-up**").anonymous().anyRequest()
                .hasRole("USER").and().formLogin().loginPage("/login")
                .defaultSuccessUrl("/", true).permitAll().and().rememberMe();
    }

    @Override
    public void configure(WebSecurity web) throws Exception {
        web.ignoring().antMatchers("/resources/**");
    }

}

File: src/main/kotlin/com/flexpoker/config/SecurityConfig.kt
package com.flexpoker.config

import com.flexpoker.login.repository.LoginRepository
import com.flexpoker.util.passwordEncoder
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.security.web.SecurityFilterChain
import javax.inject.Inject

@Configuration
@EnableWebSecurity
class SecurityConfig @Inject constructor(private val loginRepository: LoginRepository) {

    @Bean
    fun passwordEncoder(): PasswordEncoder {
        return passwordEncoder
    }

    @Bean
    fun httpSecurity(http: HttpSecurity): SecurityFilterChain {
        http.authorizeHttpRequests {
            it.requestMatchers(
                "/login",
                "/sign-up",
                "/sign-up-confirm",
                "/confirm-sign-up",
                "/userinfo",
            ).permitAll()
            it.requestMatchers("/resources/**").permitAll()
            it.anyRequest().authenticated()
        }.formLogin {
            it.loginPage("/login").permitAll()
        }.csrf { it.disable() }
        return http.build()
    }

}
